README.md

```diff
@@ -1,2 +1,100 @@
-# nbde_client
-Ansible role for configuring Network Bound Disk Encryption clients (e.g. clevis)
+nbde_client
+-----------
+
+Ansible role for configuring Network-Bound Disk Encryption clients (e.g. clevis).
+
+This role currently supports `clevis` as a provider and it uses it for operations like encryption
+and decryption.
+
+
+Supported Distributions
+-----------------------
+* RHEL-7+, CentOS-7+
+* Fedora
+
+
+Limitations
+-----------
+This role can currently create `tang` bindings. TPM2 is not supported as of now.
+
+
+Role Variables
+--------------
+
+These are the variables that can be passed to the role:
+
+
+| **Variable** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+| `nbde_client_provider` | `clevis`| identifies the provider for the `nbde_client` role. We currently support `clevis`.|
+| `nbde_client_bindings` | | a list containing binding configurations, which include e.g. devices and slots. |
+
+
+#### nbde_client_bindings
+`nbde_client_bindings` is a list of dictionaries that support the following keys:
+| **Name** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+| `device` | | specifies the path of the backing of an encrypted device in the managed host. This device must be already configured as a LUKS device before using the role (**REQUIRED**). |
+| `passphrase` | | a valid passphrase for opening/unlocking the specified device. |
+| `keyfile` | | either the absolute or relative path, in the control node, of a keyfile valid for opening/unlocking the specified device. |
```

Comment:
`key_file`, for consistency with storage. Also, I believe "on the control node" looks better.